ui/src/app/services/server-config.service.ts

```diff
@@ -0,0 +1,24 @@
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {ReplaySubject} from 'rxjs/ReplaySubject';
+
+import {ErrorHandlingService} from 'app/services/error-handling.service';
+
+import {ConfigResponse, ConfigService} from 'generated';
+
+@Injectable()
+export class ServerConfigService {
+  private configObs: Observable<ConfigResponse>;
+  constructor(
+    private configService: ConfigService, private errorHandlingService: ErrorHandlingService) {}
+
+  public getConfig(): Observable<ConfigResponse> {
+    if (!this.configObs) {
+      // Use of a replaySubject() caches the output of the API call across subscriptions.
+      const subject = new ReplaySubject<ConfigResponse>();
+      this.errorHandlingService.retryApi(this.configService.getConfig()).subscribe(subject);
+      this.configObs = subject;
+    }
+    return this.configObs;
```

Comment:
`return this.configObs.asObservable()` - I think you'll have the same issues as with `share` without explicitly calling `asObservable`.  The way I think of it is that a `Subject` is a kind of mutable `Observable`, and `asObservable` will give you a copy of the Subject as a data source without exposing the possibility of mutating the actual source (by, say, unsubscription).

Reply:
Done - good catch again. I did test for the possibility of a `refCount`-like situation and that wasn't a problem as it was for `share`. However, the caller could certainly do something unfortunate with this like calling `next()` or `unsubscribe`.

Also added test cases for this service.